app.py

```diff
@@ -41,22 +41,92 @@ def teams_create():
     return render_template('teams/create.html', pokemon=DATABASE.get("pokemon", []))
 
 # Searches for pokemon and teams in DATABASE and displays in search.html using Jinja (no AJAX)
-# - Searches should be case insensitive
-# - You should not use JavaScript or AJAX in search.html
-# - For pokemon, items should be ranked higher if any of the words in the search query are in the name, description, or types
-# - For teams, items should be ranked higher if any of the words in the search query are in the name or description
-# - Not all properties are treated equally: if there is a match in the name, rank the item higher than other properties
+# (done) Searches should be case insensitive
+# (done) You should not use JavaScript or AJAX in search.html
+# (done) For pokemon, items should be ranked higher if any of the words in the search query are in the name, description, or types
+# (done) For teams, items should be ranked higher if any of the words in the search query are in the name or description
+# (done) Not all properties are treated equally: if there is a match in the name, rank the item higher than other properties
 
 # Extra requirements when you're done:
-# - Teams should rank higher if their member pokemon also match
+# (done) Teams should rank higher if their member pokemon also match
 # - Highlight the matching word in the name or description in the search results
 # - Implement a filter for search queries (e.g. searching "p type:fire" will only
 #   search for pokemon that contain the letter p that are fire type pokemon)
 # - Add pagination to your search results page so only 20 pokemon show up at a time,
 #   but you can navigate to another page
+
+# takes a string like "p type:electric" and splits it into p, type, and electric
+def search_parser(search_string):
+        filters = re.match(r'^(?P<str>.*)\s(?P<mod>.*)[:](?P<fil>.*)(\s|$)', search_string)
+        if filters != None:
+            return filters
+        else:
+            return False
+
+# generator function that returns at the first match or yields until it's called again
+def search_db(search_string, search_object):
+    if search_string in search_object['name'].lower():
+        return search_object
+    else:
+        yield
+    if search_string in search_object['description'].lower():
+        return search_object
+    else:
+        yield
+    if 'types' in search_object.keys():
+        for t in search_object['types']:
+            if search_string == t.lower():
+                return search_object
+        return False
+    else:
+        for p in search_object['members']:
+            if search_string == pokemon[p['pokemon_id']-1]['name'].lower():
```

Comment:
With `pokemon[p['pokemon_id']-1]` what if the Pokemon weren't in the order of their IDs (e.g. a record gets deleted from the database)? Consider changing this so you're not making assumptions about the order of the `pokemon` array.